File: sdk/client-ts/src/___PLAYGROUND___/signing/sign-typed-data.ts
import { begin } from '../helpers';

export const signTypedData = async () => {
  const context = await begin();

  console.log('Signing typed data...');

  const domain = {
    name: 'Test App',
    version: '1',
    chainId: 31337,
    verifyingContract:
      '0x1234567890123456789012345678901234567890' as `0x${string}`,
  };

  const types = {
    Person: [
      { name: 'name', type: 'string' },
      { name: 'wallet', type: 'address' },
    ],
  };

  const value = {
    name: 'Alice',
    wallet: context.relayerInfo.address,
  };

  const typedData = {
    domain,
    types,
    primaryType: 'Person' as const,
    message: value,
  };

  const signature = await context.relayer.sign.typedData(typedData);

  console.log('Domain:', domain);
  console.log('Types:', types);
  console.log('Value:', value);
  console.log('Signature:', signature);

  await context.end();
};

signTypedData().then(() => console.log('sign-typed-data done'));

File: sdk/python/src/__PLAYGROUND__/signing/sign-typed-data.py
import asyncio

from __PLAYGROUND__.helper import begin, end


async def signTypedData():
    rrelayer_node = None
    try:
        _, relayer, info, rrelayer_node = await begin()

        print("Signing typed data...")

        typed_data = {
            "types": {
                "Person": [
                    {"name": "name", "type": "string"},
                    {"name": "wallet", "type": "address"},
                ],
            },
            "primaryType": "Person",
            "domain": {
                "name": "Test App",
                "version": "1",
                "chainId": 31337,
                "verifyingContract": "0x1234567890123456789012345678901234567890",
            },
            "message": {"name": "Alice", "wallet": info["address"]},
        }

        signature = await relayer.sign.typedData(typed_data)

        print("Domain:", typed_data["domain"])
        print("Types", typed_data["types"])
        print("Value", typed_data["message"])
        print("Signature", signature)

    except Exception as e:
        print("signText failed:", e)
    finally:
        if rrelayer_node is not None:
            end(rrelayer_node)


if __name__ == "__main__":
    asyncio.run(signTypedData())
    print("sign-typed-data done")
